File: src/utils/clipboard.ts
import { $ } from "zx";

export async function copyToClipboard(text: string): Promise<boolean> {
  try {
    switch (Deno.build.os) {
      case "darwin":
        await $`echo ${text} | pbcopy`.quiet();
        break;
      case "linux":
        // Try xclip first, fall back to xsel
        try {
          await $`echo ${text} | xclip -selection clipboard`.quiet();
        } catch {
          await $`echo ${text} | xsel --clipboard --input`.quiet();
        }
        break;
      case "windows":
        await $`echo ${text} | clip`.quiet();
        break;
      default:
        return false;
    }
    return true;
  } catch {
    return false;
  }
}

File: internal/cli/clipboard.go
package cli

import "context"

func copyToClipboard(ctx context.Context, runner Runner, goos, text string) bool {
	payload := text + "\n"

	switch goos {
	case "darwin":
		return runner.RunWithInput(ctx, payload, "pbcopy") == nil
	case "linux":
		if runner.RunWithInput(ctx, payload, "xclip", "-selection", "clipboard") == nil {
			return true
		}

		return runner.RunWithInput(ctx, payload, "xsel", "--clipboard", "--input") == nil
	case "windows":
		return runner.RunWithInput(ctx, payload, "clip") == nil
	}

	return false
}
